src/main/java/com/capstone/doctorsOffice/services/PrescriptionServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+package com.capstone.doctorsOffice.services;
+
+import com.capstone.doctorsOffice.dtos.PrescriptionDto;
+import com.capstone.doctorsOffice.entities.Prescription;
+import com.capstone.doctorsOffice.repositories.PatientRepository;
+import com.capstone.doctorsOffice.repositories.PrescriptionRepository;
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class PrescriptionServiceImpl implements PrescriptionService {
+    @Autowired
+    private PrescriptionRepository prescriptionRepository;
+    @Autowired
+    private PatientRepository patientRepository;
+
+    @Override
+    @Transactional
+    public List<String> addPrescription(PrescriptionDto prescriptionDto){
+        List<String> response = new ArrayList<>();
+        Prescription prescription = new Prescription(prescriptionDto);
+        prescriptionRepository.saveAndFlush(prescription);
+        response.add("New Prescription Added");
+        return response;
+    }
+
+    @Override
+    public List<String> addNewPrescription(PrescriptionDto prescriptionDto) {
+        return null;
+    }
+
```

Comment:
The `addNewPrescription(PrescriptionDto prescriptionDto)` is not doing anything. remove this one.